components/Swap/Withdraw/Wallet/SolanaWalletWithdraw.tsx

```diff
@@ -0,0 +1,177 @@
+import { Link, ArrowLeftRight } from 'lucide-react';
+import { FC, useCallback, useState } from 'react'
+import SubmitButton from '../../../buttons/submitButton';
+import toast from 'react-hot-toast';
+import { PublishedSwapTransactionStatus } from '../../../../lib/layerSwapApiClient';
+import { useSwapDataState } from '../../../../context/swap';
+import { useSettingsState } from '../../../../context/settings';
+import { Transaction, Connection, PublicKey, TransactionInstruction, TransactionResponse } from '@solana/web3.js';
+import useWallet from '../../../../hooks/useWallet';
+import { useWallet as useSolanaWallet } from '@solana/wallet-adapter-react';
+import { createAssociatedTokenAccountInstruction, createTransferInstruction, getAccount, getAssociatedTokenAddress } from '@solana/spl-token';
+import { SignerWalletAdapterProps } from '@solana/wallet-adapter-base';
+import { useSwapTransactionStore } from '../../../../stores/swapTransactionStore';
+
+type Props = {
+    depositAddress: string,
+    amount: number
+}
+
+const SolanaWalletWithdrawStep: FC<Props> = ({ depositAddress, amount }) => {
+    const [loading, setLoading] = useState(false);
+    const [transferDone, setTransferDone] = useState<boolean>();
+    const { getWithdrawalProvider } = useWallet()
+
+    const { setSwapTransaction } = useSwapTransactionStore();
+    const { swap } = useSwapDataState();
+
+    const { networks, layers } = useSettingsState();
+    const { source_network: source_network_internal_name } = swap || {};
+    const source_network = networks.find(n => n.internal_name === source_network_internal_name);
+    const source_layer = layers.find(l => l.internal_name === source_network_internal_name)
+    const source_currency = source_network?.currencies?.find(c => c.asset.toLocaleUpperCase() === swap?.source_network_asset.toLocaleUpperCase());
+
+    const provider = getWithdrawalProvider(source_layer!);
+    const wallet = provider?.getConnectedWallet();
+    const { publicKey: walletPublicKey, signTransaction } = useSolanaWallet();
+    const solanaNode = source_network?.nodes[0].url;
+
+    const handleTransaction = async (swapId: string, publishedTransaction: TransactionResponse, txHash: string) => {
+        if (publishedTransaction?.meta?.err) {
+            txHash && setSwapTransaction(swapId, PublishedSwapTransactionStatus.Error, txHash, String(publishedTransaction.meta.err));
+            toast(String(publishedTransaction.meta.err))
+            setLoading(false)
+        }
+        else {
+            txHash && setSwapTransaction(swapId, PublishedSwapTransactionStatus.Completed, txHash, '');
```

Comment:
This else will never be executed because before calling this handleTransaction method you have this !txReceipt?.meta?.err check